### PR TITLE
fix: Vote segmentation

### PR DESCRIPTION
**Pull request by ncomerci**
**Screenshots**
image

- [x]  **Functionality validation**
    - [x]  Feature working as described in the related issue
    - [x]  Code errors are being logged in the code health service
- [x]  **Product validation**
    - [x]  Preview link if it’s a customer facing issue

### CODE REVIEW

File: src/hooks/useVotesCountByDate.ts [hunk 1]
```diff
@@ -13,8 +14,15 @@ export default function useVotesCountByDate(start: Date, end: Date) {
...
+  const parsedVotes = votes ? createMultipleVotes(votes) : undefined
+  const { highQualityVotes } = getMultipleVotesSegmentation(parsedVotes)
+
+  const votesCount = Object.values(highQualityVotes).reduce((acc, votes) => {
+    return acc + votes.length
+  }, 0)
+
   return {
-    votesCount: votes?.length,
+    votesCount: votes ? votesCount : undefined,
```
Comment (andyesp) [inline src/hooks/useVotesCountByDate.ts:25]: Do we need `votesCount` to be undefined? If highQualityVotes is an empty array, votesCount will return 0.

File: src/entities/Votes/utils.ts [hunk 2]
```diff
@@ -163,3 +182,49 @@ export function abbreviateNumber(vp: number) {
 function getFloorOrZero(number?: number) {
   return Math.floor(number || 0)
 }
+
+export function getVoteSegmentation(votes: Record<string, Vote> | null | undefined): VoteSegmentation<Vote> {
+  const highQualityVotes: Record<string, Vote> = {}
```
Comment (andyesp) [inline src/entities/Votes/utils.ts:187]: Maybe this `Record<string, Vote>` type could have a name that can help us name the functions return this type 🤔

File: src/entities/Votes/utils.ts [hunk 3]
```diff
@@ -163,3 +182,49 @@ export function abbreviateNumber(vp: number) {
...
+  return {
+    highQualityVotes,
+    lowQualityVotes,
+  }
+}
+
+export function getMultipleVotesSegmentation(
+  votes: Record<string, Vote[]> | null | undefined
+): VoteSegmentation<Vote[]> {
+  const highQualityVotes: Record<string, Vote[]> = {}
```
Comment (andyesp) [inline src/entities/Votes/utils.ts:209]: Same as above comemnt

File: src/entities/Votes/utils.ts [hunk 4]
```diff
@@ -16,20 +18,38 @@ export function toProposalIds(ids?: undefined | null | string | string[]) {
   return list.filter((id) => isUUID(String(id)))
 }
 
-export function createVotes(votes: SnapshotVote[]) {
+function _createVotes<T>(votes: SnapshotVote[], returnValue: (vote: Vote, prevValue?: T) => T): Record<string, T> {
```
Comment (andyesp) [inline src/entities/Votes/utils.ts:21]: This function could be named just `createVotes` as it is no longer clashing with the other function name.

**Review by andyesp — APPROVED** [review]

File: src/pages/proposal.tsx [hunk 5]
```diff
@@ -430,7 +404,7 @@ export default function ProposalPage() {
           )}
           {showSurveyResults && (
             <SurveyResults
-              votes={votes}
+              votes={highQualityVotes ?? null}
```
Comment (1emu) [inline src/pages/proposal.tsx:407]: Why are we including only the high vp votes for the survey results? 
Are bots also sending a survey to snapshot?

Comment (1emu) [reply src/pages/proposal.tsx:407]: I guess it makes sense if we only count the high quality votes for displaying the proposal's partial results...

**Review by 1emu — APPROVED** [review]

**Pull request merged**